code/game/machinery/portable_turret.dm

```diff
@@ -511,61 +379,83 @@ Status: []<BR>"},
 		popDown()
 		return
 
+	last_fired = 0
+	if(attacked)
+		attacked--
+
 	var/list/targets = list()		   // list of primary targets
 	var/list/secondarytargets = list() // targets that are least important
 
-	if(src.check_anomalies) // if its set to check for xenos/carps, check for non-mob "crittersssss"(And simple_animals)
-		for(var/mob/living/simple_animal/C in view(7,src))
-			if(C.stat)
+
+
+	if(emagged)
```

Comment:
Instead of copy pasting a bunch of stuff, make the regular (non-emagged) loop through all mobs in view (size determined by whether it's emagged or not), and then change the logic for whether the target is valid or not based on whether it's emagged.